HIP/hip-1056.md

```diff
@@ -0,0 +1,1933 @@
+  * from TransactionBody.
+  */
+  uint64 transaction_fee_charged = 5;
+
+  /**
+  * All hbar transfers as a result of this transaction, such as fees, or
+  * transfers performed by the transaction, or by a smart contract it calls, or
+  * by the creation of threshold records that it triggers.
+  */
+  proto.TransferList transfer_list = 6;
+
+  /**
+  * All Token transfers as a result of this transaction.
+  */
+  repeated proto.TokenTransferList token_transfer_lists = 7;
+
+  /**
+  * All token associations implicitly created while handling this transaction.
+  */
+  repeated proto.TokenAssociation automatic_token_associations = 8;
+
+  /**
+  * List of accounts with the corresponding staking rewards paid as a result of
+  * a transaction.
+  */
+  repeated proto.AccountAmount paid_staking_rewards = 9;
+
+  /**
+  * Congestion pricing multiplier at the time the transaction was executed.
+  */
+  uint64 congestion_pricing_multiplier = 10;
+}
+```
+
+#### Transaction Output
+
+Output data from a transaction data that is neither present in the transaction nor stored in state. Not all transactions
+have output, each one that does has a corresponding output message type with any data it needs to include.
+
+
+```protobuf
+message TransactionOutput {
+  oneof transaction {
+    /**
+    * Output from a crypto transfer transaction.
+    */
+    CryptoTransferOutput crypto_transfer = 1;
+
+    /**
+    * Output from a utilPrng transaction to request a
+    * deterministic pseudo-random number.
+    */
+    UtilPrngOutput util_prng = 2;
+
+    /**
+    * Output from a contract call transaction.
+    */
+    CallContractOutput contract_call = 3;
+
+    /**
+    * Output from an ethereum call transaction.
+    */
+    EthereumOutput ethereum_call = 4;
+
+    /**
+    * Output from a contract create transaction.
+    */
+    CreateContractOutput contract_create = 5;
+
+    /**
+    * Output from a schedule create transaction that executed
+    * immediately on creation.
+    */
+    CreateScheduleOutput create_schedule = 6;
+
+    /**
+    * Output from a schedule sign transaction that resulted in
+    * executing the scheduled transaction.
+    */
+    SignScheduleOutput sign_schedule = 7;
+  }
+}
+```
+
+```protobuf
+message CryptoTransferOutput {
+  /**
+  * Custom fees assessed during a CryptoTransfer.
+  * <p>
+  * These fees SHALL be present in the full transfer list for the transaction.
+  */
+  repeated proto.AssessedCustomFee assessed_custom_fees = 1;
+}
+```
+
+```protobuf
+message UtilPrngOutput {
+  oneof entropy {
+    /**
+    * A deterministic pseudo-random sequence of 48 bytes.
+    */
+    bytes prng_bytes = 1;
+
+    /**
+    * A deterministic pseudo-random number generated within a
+    * specified range.
+    */
+    uint32 prng_number = 2;
+  }
+}
+```
+
+```protobuf
+message CallContractOutput {
+  /**
+  * A list of additional outputs
+  */
+  repeated proto.TransactionSidecarRecord sidecars = 1;
```

Comment:
Agreed on the flattening.
We'll take a look

Reply:
Maybe something like this

```protobuf
message CallContractOutput {
     /**
     * A result for an Ethereum Transaction execution, either a create or call.
     * <p>
     * This field SHALL contain all of the data produced by the contract
     * create transaction as well as basic accounting results.
     */
     proto.ContractFunctionResult contract_function_result = 1;
 
     /**
     * The storage changes to a smart contract's storage as a side effect of the function call.
     */
     repeated ContractStateChanges state_changes = 2;
 
     /**
     * The intercontract interaction details. This represents teh internal EVM message frames.
     */
     repeated ContractActions contract_actions = 3;
}

message CreateContractOutput {
     /**
     * A result for an Ethereum Transaction execution, either a create or call.
     * <p>
     * This field SHALL contain all of the data produced by the contract
     * create transaction as well as basic accounting results.
     */
     proto.ContractFunctionResult contract_function_result = 1;
 
     /**
     * Contract bytecode during deployment
     */
     bytes initcode = 3;
 
     /**
     * Contract bytecode after deployment
     */
     bytes runtime_bytecode = 4;
 
     /**
     * The storage changes to a smart contract's storage as a side effect of the function call.
     */
     repeated ContractStateChanges state_changes = 5;
 
     /**
     * The intercontract interaction details. This represents teh internal EVM message frames.
     */
     repeated ContractActions actions = 6;
}

message EthereumOutput {
    /**
     * An ethereum hash value.
     * <p>
     * This SHALL be a keccak256 hash of the ethereumData.
     */
    bytes ethereum_hash = 1;

    /**
    * A result for an Ethereum Transaction execution, either a create or call.
    * <p>
    * This field SHALL contain all of the data produced by the contract
    * create transaction as well as basic accounting results.
    */
    proto.ContractFunctionResult contract_function_result = 2;

    /**
    * Contract bytecode during deployment
    */
    bytes initcode = 3;

    /**
    * Contract bytecode after deployment
    */
    bytes runtime_bytecode = 4;

    /**
    * The storage changes to a smart contract's storage as a side effect of the function call.
    */
    repeated ContractStateChanges state_changes = 5;

    /**
    * The intercontract interaction details. This represents teh internal EVM message frames.
    */
    repeated ContractActions actions = 6;
}
```

Reply:
@steven-sheehy , @jasperpotts and @jsync-swirlds pushed an improved layout for HSCS. Should be flatter, reuses where possible and ensure output types only contain info applicable to their transaction type

Reply:
Got a little delayed on this, but is it necessary to duplicate the contract storage changes here?  Those should all be in the `StateChange` items (which follow the outputs).

Reply:
Good point, If we can remove it because state changes captures it that's great, only problem is `ContractStateChange` calls out the contract_id. Can we infer that? If so then we can remove this

Reply:
Moving this to open issues so it's not missed